cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -172,15 +183,36 @@ def _add_volume_from_secret(secret: Dict[Text, Text]):
     )
 
 
-def delete_old_cron_jobs(repo_name: Text, dry_run: bool = False):
+@toolz.curry
+def _delete_cron_job(api_instance: client.BatchV1beta1Api, dry_run: bool, name: Text):
+    options = {"name": name, "namespace": "default"}
+    _set_dry_run(options, dry_run)
+    api_instance.delete_namespaced_cron_job(**options)
+
+
+def delete_old_cron_jobs(repo_name: Text, new_jobs, dry_run: bool = False):
     api_instance = client.BatchV1beta1Api()
     options = {"namespace": "default", "label_selector": f"repository={repo_name}"}
-    if dry_run:
-        options["dry_run"] = "All"
-    api_instance.delete_collection_namespaced_cron_job(**options)
-    logging.info("Deleting CronJobs. Waiting 2 min for deletion to complete...")
-    # TODO(Erez): Find an event driven way to wait for this
-    time.sleep(2 * 60)
+    api_response = api_instance.list_namespaced_cron_job(**options)
+    gamla.pipe(
+        api_response.items,
+        curried.map(lambda cronjob: cronjob.metadata.name),
+        curried.filter(
+            lambda cronjob_name: cronjob_name
```

Comment:
use `operator.contains` instead of this lambda (import the curried version from toolz.curried)